posthog/api/insight.py

```diff
@@ -1055,6 +1061,21 @@ def calculate_trends(self, request: request.Request) -> dict[str, Any]:
 
         return {"result": result, "timezone": team.timezone}
 
+    @cached_by_filters
+    def calculate_trends_hogql(self, request: request.Request) -> dict[str, Any]:
+        team = self.team
+        filter = Filter(request=request, team=team)
+        query = filter_to_query(filter.to_dict())
+        query_runner = get_query_runner(query, team, limit_context=None)
+
+        # we use the legacy caching mechanism (@cached_by_filters decorator), no need to cache in the query runner
+        result = query_runner.run(execution_mode=ExecutionMode.CALCULATE_BLOCKING_ALWAYS)
```

Comment:
Where do customers call this endpoint from and does this have the same rate limit as `/query`?
As `CALCULATE_BLOCKING_ALWAYS` gives them a way around async processing.

Reply:
They call `https://us.posthog.com/api/projects/:team_id/insights/trend/`, and yes this allows bypassing async processing and always has (legacy endpoint didn't have async processing).

I think we'll eventually bug them to use the new `/query` api, but this allows us to get rid of legacy code while giving them a longer grace period.

Current usage can be seen on this dashboard https://us.posthog.com/project/2/dashboard/289911 e.g. the funnels endpoint is only used by one customer.